packages/jest-types/__typetests__/jest.test.ts

```diff
@@ -7,89 +7,320 @@
+expectError(jest.dontMock());
+
+expectType<typeof jest>(jest.disableAutomock());
+expectError(jest.disableAutomock(true));
+
 expectType<typeof jest>(jest.enableAutomock());
+expectError(jest.enableAutomock('moduleName'));
+
+expectType<typeof jest>(jest.isolateModules(() => {}));
+expectError(jest.isolateModules());
+
 expectType<typeof jest>(jest.mock('moduleName'));
 expectType<typeof jest>(jest.mock('moduleName', jest.fn()));
 expectType<typeof jest>(jest.mock('moduleName', jest.fn(), {}));
 expectType<typeof jest>(jest.mock('moduleName', jest.fn(), {virtual: true}));
-expectType<typeof jest>(jest.resetModules());
-expectType<typeof jest>(jest.isolateModules(() => {}));
-expectType<typeof jest>(jest.retryTimes(3));
-expectType<Mock<Promise<string>, []>>(
-  jest
-    .fn(() => Promise.resolve('string value'))
-    .mockResolvedValueOnce('A string, not a Promise'),
+expectError(jest.mock());
+
+expectType<typeof jest>(jest.unstable_mockModule('moduleName', jest.fn()));
+expectType<typeof jest>(
+  jest.unstable_mockModule('moduleName', () => Promise.resolve(jest.fn())),
 );
-expectType<Mock<Promise<string>, []>>(
-  jest
-    .fn(() => Promise.resolve('string value'))
-    .mockResolvedValue('A string, not a Promise'),
+expectType<typeof jest>(jest.unstable_mockModule('moduleName', jest.fn(), {}));
+expectType<typeof jest>(
+  jest.unstable_mockModule('moduleName', () => Promise.resolve(jest.fn()), {}),
 );
-expectType<Mock<Promise<string>, []>>(
-  jest
-    .fn(() => Promise.resolve('string value'))
-    .mockRejectedValueOnce(new Error('An error, not a string')),
+expectType<typeof jest>(
+  jest.unstable_mockModule('moduleName', jest.fn(), {virtual: true}),
 );
-expectType<Mock<Promise<string>, []>>(
-  jest
-    .fn(() => Promise.resolve('string value'))
-    .mockRejectedValue(new Error('An error, not a string')),
+expectType<typeof jest>(
+  jest.unstable_mockModule('moduleName', () => Promise.resolve(jest.fn()), {
+    virtual: true,
+  }),
 );
 
-expectType<void>(jest.runAllImmediates());
-expectType<void>(jest.runAllTicks());
-expectType<void>(jest.runAllTimers());
-expectType<void>(jest.runOnlyPendingTimers());
-expectType<void>(jest.advanceTimersByTime(9001));
+expectType<unknown>(jest.requireActual('./pathToModule'));
+expectError(jest.requireActual());
+
+expectType<unknown>(jest.requireMock('./pathToModule'));
+expectError(jest.requireMock());
+
+expectType<typeof jest>(jest.resetModules());
+expectError(jest.resetModules('moduleName'));
 
-expectType<typeof jest>(jest.setMock('moduleName', {}));
-expectType<typeof jest>(jest.setMock('moduleName', {}));
 expectType<typeof jest>(jest.setMock('moduleName', {a: 'b'}));
-expectType<typeof jest>(jest.setTimeout(9001));
+expectError(jest.setMock('moduleName'));
+
 expectType<typeof jest>(jest.unmock('moduleName'));
-expectType<typeof jest>(jest.useFakeTimers());
-expectType<typeof jest>(jest.useRealTimers());
+expectError(jest.unmock());
+
+// Mock Functions
```

Comment:
should we have these tests in `jest-mock`, and just have an `expectType<JestMock['fn']>(jest.fn)` and `expectType<JestMock['isMockFunction']>(jest.isMockFunction)`? Gut feeling is that's more natural since at the `jest.*` level it's more of an integration test and not a "unit".

Of course, doesn't really matter, but the tests would be closer to the code it tests

Reply:
That's good idea. I will move them but later, not at computer right now. Also tests for Mock Function fits better in `jest-mock`.